zrcs/zrc-3.md

```diff
@@ -0,0 +1,431 @@
+transition RevokeDefaultTeller(teller : ByStr20)
+```
+
+|        | Name        | Type      | Description                                          |
+| ------ | ----------- | --------- | ---------------------------------------------------- |
+| @param | `teller`      | `ByStr20` | Address of the default teller to be revoked.   |
+
+
+|           | Name          | Description                | Event Parameters                                                                                                                                                                                                                   |
+| --------- | ------------- | -------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| eventName | `RevokedDefaultTellerSuccess` | Revoking is successful.     | `teller`: `ByStr20`, `recipient`: `ByStr20`, and `sender` : `_sender`.                             |
+| eventName | `Error`       | Revoking is not successful. | - emit `CodeNotFound` if the default teller is not found. |
+
+
+#### 3. Send
+
+```ocaml
+(* @dev: Moves amount tokens from the caller’s address to the recipient.   *)
+(* @param from:       Address of the sender whose balance is decreased.    *)
+(* @param recipient:  Address of the recipient whose balance is increased. *)
+(* @param amount:     Amount of tokens to be sent.                         *)
+transition Send(from: ByStr20, recipient: ByStr20, amount: Uint128)
+```
+
+|        | Name        | Type      | Description                                          |
+| ------ | ----------- | --------- | ---------------------------------------------------- |
+| @param | `from`      | `ByStr20` | Address of the sender whose balance is decreased.    |
+| @param | `recipient` | `ByStr20` | Address of the recipient whose balance is increased. |
+| @param | `amount`    | `Uint128` | Amount of tokens to be sent.                         |
+
+|           | Name          | Description                | Event Parameters                                                                                                                                                                                                                   |
+| --------- | ------------- | -------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| eventName | `SendSuccess` | Sending is successful.     | `from`: `ByStr20`, `recipient`: `ByStr20`, and `amount`: `Uint128`.                             |
+| eventName | `Error`       | Sending is not successful. | - emit `CodeNotAuthorised` if the transition is called by a user who is not the contract owner. |
```

Comment:
I think this is being borrowed from ZRC-2 but seems like this is wrong. The caller of  `Send` should not be the contract owner. Otherwise, it becomes the super default owner who can transfer any number of tokens around for anyone.

Reply:
Yep, good catch. Updated that in the ZRC-2.